backend/sirius-web-spring-collaborative-trees/src/main/java/org/eclipse/sirius/web/spring/collaborative/trees/TreeEventProcessor.java

```diff
@@ -121,16 +122,29 @@ public ISubscriptionManager getSubscriptionManager() {
     }
 
     @Override
-    public void refresh() {
-        long start = System.currentTimeMillis();
+    public void refresh(ChangeKind changeKind) {
+        if (this.shouldRefresh(changeKind)) {
+            long start = System.currentTimeMillis();
 
-        Tree tree = this.refreshTree();
+            Tree tree = this.refreshTree();
 
-        this.currentTree.set(tree);
-        this.sink.next(new TreeRefreshedEventPayload(tree));
+            this.currentTree.set(tree);
+            this.sink.next(new TreeRefreshedEventPayload(tree));
+
+            long end = System.currentTimeMillis();
+            this.timer.record(end - start, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private boolean shouldRefresh(ChangeKind changeKind) {
+        boolean shouldRefresh = false;
+
+        shouldRefresh = shouldRefresh || changeKind == ChangeKind.SEMANTIC_CHANGE;
+        shouldRefresh = shouldRefresh || changeKind == ChangeKind.REPRESENTATION_CREATION;
+        shouldRefresh = shouldRefresh || changeKind == ChangeKind.REPRESENTATION_DELETION;
+        shouldRefresh = shouldRefresh || changeKind == ChangeKind.REPRESENTATION_RENAMING;
```

Comment:
A switch on the value of `changeKind` would be clearer IMO, and ensure all the values are considered.

Reply:
I'm not sure that all values need to be considered in details but it would indeed be clearer